docs/integrating-jupyter.md

```diff
@@ -0,0 +1,342 @@
+```
+
+## 🚧 Integrated compute with pyodide via JupyterLite
+
+The [JupyterLite](https://jupyterlite.readthedocs.io/en/latest/) server and `pyodide` kernels can be activated using:
+
+```{code-block} yaml
+---
+caption: minimal configuration for enabling JupyterLite
+---
+project:
+    thebe:
+        lite: true
+```
+
+```{important} TODO
+Add the specific list options for custom wheel paths, etc...
+```
+
+## 🚧 Local Development Mode
+
+When working on a MyST site using the `myst-cli`, using a local Jupyter server connection makes a lot of sense and speeds up development. The `local` key allows you to enable and configure a local environment without having to change the other (remote) settings in your `myst.yml` file that will be used in your final deployment.
+
+Local development using can be enabled by simply adding the `local` key, which will use default server options.
+
+```yaml
+---
+caption: minimal configuration of a local development connection
+---
+project:
+  github: https://github.com/executablebooks/thebe-binder-base
+    thebe:
+        binder: true
+        local: true
+```
+
+Further configure the `local` connection using the following options.
+
+```{list-table}
+:header-rows: 1
+
+* - `key`
+  - desciption
+  - default
+* - `url`
+  - The base url of the Jupyter server you want to connect to
+  - `http://localhost:8888`
+* - `token`
+  - The token needed to establish a connection
+  - `test-secret`
+* - `kernelName`
+  - The name of the kernel to request when stating a session
+  - `python`
+```
+
+### Start a local Jupyter server
+
+In addition to how you might normally start a JupyterLab session, it's necessary to provide two additional command line options, as follows.
+
+```{code} bash
+jupyter lab --NotebookApp.token=test-secret --NotebookApp.allow_origin='*'
+```
+
+The command above is fine for local development and the `token` used should align with that provided in the `project.thebe.token` key.
+
+When starting a local Jupyter server for use with MyST it's also important to understand your computational environment and ensure that the Jupyter instance has access to that with the dependencies it needs to run. This is acheived by following normal best practices for reproducible environment configuration, if you're not familiar with these see [REES](https://repo2docker.readthedocs.io/en/latest/specification.html).
+
+## Reference
+
+### Complete options schema
+
+```{code-block} yaml
+project:
+  thebe: undefined(false) | boolean | object
+    lite: boolean
+    binder: undefined(false) | boolean | object
+      url: string (url)
+      repo: string (org-name/repo-name)
+      ref: string (valid git refs only?)
+      provider: string (git | gitlab | github)
+    server:  undefined(false) | boolean | object
+      url: string (url)
+      token: string (any)
+    kernelName: string (any)
+    sessionSaving: boolean (default: true)
```

Comment:
Should this be `disableSessionSaving: boolean (default: false)`?

And also updated in the table below?